docs/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/set-up-cloud-providers/amazon.md

```diff
@@ -159,6 +163,357 @@ Setting the value of the tag to `owned` tells the cluster that all resources wit
+
+When downtime is acceptable, you can switch to external cloud provider which removes in-tree components and then deploy charts to install AWS cloud controller manager as explained in [using out of tree cloud provider](#using-out-of-tree-aws-cloud-provider).
+
+When control plane cannot tolerate downtime, leader migration must be enabled to facilitate a smooth migration from the controllers in the kube controller manager to their counterparts in the cloud controller manager. Refer to [Using leader migration](https://cloud-provider-aws.sigs.k8s.io/getting_started/) for more details.
+
+**Important**
+- [Cloud controller migration docs](https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/#before-you-begin) mentions that it is possible to migrate with the same Kubernetes version, but assumes that migration is part of Kubernetes upgrade.
+
+- Refer [Migrate to use cloud controller manager](https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/) to confirm if any customizations are required before migrating.
+  Confirm [migration configuration values](https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/#default-configuration) and special case around [migrating IPAM controllers](https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/#node-ipam-controller-migration).
+
+
+1. Update cluster config to enable leader migration in `cluster.yml`
+
+```
+services:
+  kube-controller:
+    extra_args:
+      enable-leader-migration: "true"
+```
+Note that cloud provider is still `aws` at this step.
+```
+cloud_provider:
+  name: aws
+```
+
+2. Cordon control plane nodes so aws cloud controller pods run on nodes only after upgrading to external cloud provider.
+
+```
+kubectl cordon -l "node-role.kubernetes.io/controlplane=true"
+```
+
+3. The steps to install AWS cloud controller manager for migration are same as fresh installation, but you must enable leader migration. 
+Add the following to the container arguments in step 7 when following the [steps to install chart](#helm-chart-installation):
+
+```
+- '--enable-leader-migration=true' 
+```
+
+4. Confirm the chart is installed but the new pods aren't running yet due to cordoned controlplane nodes. After updating the cluster in thenext step, rke will uncordon each node after upgrading and `aws-controller-manager` pods will be scheduled.
+
+5. Update cluster.yml to change cloud provider and remove leader migration args from kube-controller.
+
+Selecting **External Amazon (out-of-tree)** will set `--cloud-provider=external` and allow enabling `useInstanceMetadataHostname`. You must enable `useInstanceMetadataHostname` for node-driver clusters and for custom clusters if not providing custom node name via `--node-name`. Enabling `useInstanceMetadataHostname` will query ec2 metadata service and set `/hostname` as `hostname-override` for `kubelet` and `kube-proxy`:
```

Comment:
`if not providing custom node name` -> since this is basically an upgrade case where nodes are already registered. Would the user add `useInstanceMetadataHostname` by default then for custom clusters too? 

I have the following use cases that would be great to have clarity on, what should the user do if:
* they already have nodes registered without using --node-name
* they have nodes registered and did use --node-name

if we instruct them to enable it, what happens if:
* they plan on adding more nodes later on and use --node-name override?

Reply:
@kinarashah Need clarification on this

Reply:
Sorry @jakefhyde, I missed the tag notification.  

@slickwarren The main rule here is that node-name should match what the cloud provider expects. If not, users would need to use `useInstanceMetadataHostname` to let rancher query the ec2 service. So 
- they already have nodes registered without using --node-name => use `useInstanceMetadataHostname` 
- they have nodes registered and did use --node-name => if node-name matches what the cloud provider expects, they're good. If not, use `useInstanceMetadataHostname`  

- if we instruct them to enable it, what happens if: they plan on adding more nodes later on and use --node-name override? => this is tricky, but since the logic is at cluster level, they'd need to continue using whatever approach they used for existing nodes unless the existing nodes are reregistered the same way as new nodes